fix_proto_conf_desc.py

```diff
@@ -0,0 +1,177 @@
+
+    def get_size(self):
+        return self.tif.get_unpadded_size()
+
+class PCDStruct(CStruct): 
+    PCD_BASE_IDA_STRUCT_TYPE_NAME = "ProtConfDescriptor"
+
+    def _create_pcd(self, pcd_address):
+        self.tif = ida_typeinf.tinfo_t()
+        udt = ida_typeinf.udt_type_data_t()
+        self.tif.create_udt(udt)
+        self.tif.set_named_type(None, self.PCD_BASE_IDA_STRUCT_TYPE_NAME+"_"+hex(pcd_address))
+        self.append_reloffset_member("ProtocolDescriptor")
+        self.append_reloffset_member("NominalTypeDescriptor")
+        #If PWT field creates a reloffset to itself (PWT dword = 0) it means it doesn't have PWT.
+        self.append_reloffset_member("PWT")
+        self.append_int32_member("Flags")
+
+    def __init__(self, pcd_address):  
+        self._create_pcd(pcd_address)
+
+class RelWitnessTableStruct(CStruct):
+    RWT_BASE_IDA_STRUCT_TYPE_NAME = "RelWitTable"
+
+    def _create_rwt(self, rwt_addr, rel_wit_num):
+        self.tif = ida_typeinf.tinfo_t()
+        udt = ida_typeinf.udt_type_data_t()
+        self.tif.create_udt(udt)
+        self.tif.set_named_type(None, self.RWT_BASE_IDA_STRUCT_TYPE_NAME+"_"+hex(rwt_addr))
+        self.append_int32_member("NumWitnesses")
+        rwt_addr += 4
+        for idx in range(rel_wit_num):
+            req_addr = rwt_addr+(idx*2*4)
+            req_name = get_symbol_name_from_address(req_addr)
+            self.append_reloffset_member("req_"+req_name)
+            self.append_reloffset_member("impl_"+req_name)
+
+    def __init__(self, rwt_addr):
+        rel_wit_num = get_word_with_size(rwt_addr, 4)
+        self._create_rwt(rwt_addr, rel_wit_num)
+
+class GenWitnessTableStruct(CStruct):
+    GWT_BASE_IDA_STRUCT_TYPE_NAME = "GenWitTable"
+
+    def _create_gwt(self, gwt_addr):
+        self.tif = ida_typeinf.tinfo_t()
+        udt = ida_typeinf.udt_type_data_t()
+        self.tif.create_udt(udt)
+        self.tif.set_named_type(None, self.GWT_BASE_IDA_STRUCT_TYPE_NAME+"_"+hex(gwt_addr))
+        self.append_int16_member("WitnessTableSizeInWords")
+        self.append_int16_member("WitnessTablePrivateSizeInWords")
+        #If Instantiator field creates a reloffset to itself (Instantiator dword = 0) it means it doesn't have Instantiator.
+        self.append_reloffset_member("Instantiator")
+        self.append_reloffset_member("PrivateData")
+
+    def __init__(self, gwt_addr):
+        self._create_gwt(gwt_addr)
+
+def get_symbol_name_from_address(ea):
+    requirement_offset = get_word_with_size(ea, 4)
+    requirement_addr = ea + ( ((requirement_offset & 0xffffffff)^0x80000000)-0x80000000 )
+    definition_addr = get_word_with_size(requirement_addr-1, 8)
+    name = ida_name.get_name(definition_addr)
+    try:
+        return idc.demangle_name(name, idc.get_inf_attr(idc.INF_SHORT_DN)) 
+    except Exception as e:
+        #Could not demangle the symbol. Just return the address 
+        return str(requirement_addr)
+
+def get_word_with_size(ea, size):
```

Comment:
i much rather using `ida_bytes.get_wide_dword(ea)` instead since a WORD should be the CPU register size and nothing else. this just makes it more confusing